website/docs/reference/dbt-jinja-functions/cross-database-macros.md

```diff
@@ -0,0 +1,537 @@
+---
+title: "cross-database macros"
+id: "cross-database-macros"
+---
+
+# Overview
+
+These macros make it easier for package authors (especially those writing modeling packages) to implement cross-database
+compatibility. In general, you should not need these macros in your own dbt project (unless it is a package or you otherwise desire cross-database compatibility).
+
+## All functions (alphabetical)
+- [dbt.any_value](#any_value)
+- [dbt.bool_or](#bool_or)
+- [dbt.cast_bool_to_text](#cast_bool_to_text)
+- [dbt.concat](#concat)
+- [dbt.current_timestamp](#current_timestamp)
+- [dbt.current_timestamp_in_utc](#current_timestamp_in_utc)
+- [dbt.dateadd](#dateadd)
+- [dbt.datediff](#datediff)
+- [dbt.date_trunc](#date_trunc)
+- [dbt.escape_single_quotes](#escape_single_quotes)
+- [dbt.except](#except)
+- [dbt.hash](#hash)
+- [dbt.intersect](#intersect)
+- [dbt.last_day](#last_day)
+- [dbt.length](#length)
+- [dbt.listagg](#listagg)
+- [dbt.position](#position)
+- [dbt.replace](#replace)
+- [dbt.right](#right)
+- [dbt.safe_cast](#safe_cast)
+- [dbt.split_part](#split_part)
+- [dbt.string_literal](#string_literal)
+
+## Set functions
+- [dbt.except](#except)
+- [dbt.intersect](#intersect)
+
+## String functions
+- [dbt.concat](#concat)
+- [dbt.hash](#hash)
+- [dbt.length](#length)
+- [dbt.position](#position)
+- [dbt.replace](#replace)
+- [dbt.right](#right)
+- [dbt.split_part](#split_part)
+
+## Aggregate and window functions
+- [dbt.any_value](#any_value)
+- [dbt.bool_or](#bool_or)
+- [dbt.listagg](#listagg)
+
+## Cast functions
+- [dbt.cast_bool_to_text](#cast_bool_to_text)
+- [dbt.safe_cast](#safe_cast)
+
+## Non-SQL Jinja functions
```

Comment:
Are these really "Non-SQL"? I thought these macros existed to produce SQL expressions for escaped strings and string literals

Reply:
Maybe you can help come up with a more accurate name?

Your understanding is correct, and this first-draft description is somewhat of a misnomer. All of these macros _are_ used to help build up executable SQL.

### Intent
I was trying to differentiate _where_ the actual escaping or literal-ing will happen for these two macros (pre-SQL via Jinja or SQL via database):
- `escape_single_quotes`
- `string_literal`

### Why this is hard to think about
So many cross-data macros are pretty meta (since they render _other_ code that will actually do the thing described (like `pivot`, etc).

These particular macros aren't meta -- they actually do what they say they will do _directly_, without passing along the actual work to the database.

Reply:
> pre-SQL via Jinja or SQL via database

Ohhhhh. I see what you mean. Hm, maybe like "string literal" functions (as distinct from "string functions")? Makes me think of the subtleties around `try_cast` in Snowflake, which supports string literal inputs _only_